frontend/ui/src/pages/Inbox/MessageInput/InputSelector.tsx

```diff
@@ -3,71 +3,75 @@ import styles from './InputSelector.module.scss';
 import {ReactComponent as Close} from 'assets/images/icons/close.svg';
 import {SourceMessage} from 'render';
 import {Source, Message} from 'model';
-import {FileInfo} from './index';
 
 type InputSelectorProps = {
   messageType: 'template' | 'suggestedReplies' | 'message';
   message: Message;
   source: Source;
-  contentResizedHeight: number;
-  fileInfo: FileInfo | null;
   removeElementFromInput: () => void;
+  contentResizedHeight: number;
 };
 
+const textareaHeight = 40;
+const minImageHeight = 50;
+
 export const InputSelector = (props: InputSelectorProps) => {
-  const {source, message, messageType, removeElementFromInput, contentResizedHeight, fileInfo} = props;
+  const {source, message, messageType, removeElementFromInput, contentResizedHeight} = props;
   const [closeIconWidth, setCloseIconWidth] = useState('');
   const [closeIconHeight, setCloseIconHeight] = useState('');
-  const [closeButtonSelector, setCloseButtonSelector] = useState(false);
+  const [selectorPreviewCloseButton, setSelectorPreviewCloseButton] = useState(false);
 
   const fileSelectorDiv = useRef(null);
   const removeFileButton = useRef(null);
 
-  const scaleInputSelector = () => {
-    if (fileSelectorDiv?.current?.offsetHeight > contentResizedHeight) {
-      const contentSelectorDivHeight = fileSelectorDiv.current.offsetHeight;
-      const scaleRatio = Number(Math.min(contentResizedHeight / contentSelectorDivHeight).toFixed(2));
+  const resizeObserver = new ResizeObserver(entries => {
+    for (const entry of entries) {
+      const fileSelectorHeight = entry.contentRect.height;
+      const fileSelectorWidth = entry.contentRect.width;
+
+      if (fileSelectorHeight > contentResizedHeight) {
+        scaleDownInputSelector(fileSelectorHeight);
+      } else if (fileSelectorHeight >= textareaHeight && fileSelectorWidth > minImageHeight) {
+        setSelectorPreviewCloseButton(true);
+      }
+    }
+  });
```

Comment:
refactored InputSelector using `ResizeObserver` to dynamically track DOM size changes